### PR TITLE
 [FIX] sale_order_product_recommendation: wizard didn't properly get on memory created lines

### PR DESCRIPTION
**Pull request by chienandalu**
I couldn't get to the origin point of the issue (lines info gets lost in the wizard action) but this patch fixes it.

cc @Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@Yajo you were the one that creates this. Can you please check why this is needed?

File: sale_order_product_recommendation/wizards/sale_order_recommendation.py [hunk 1]
```diff
@@ -80,42 +80,37 @@ def _generate_recommendations(self):
         )
         found_dict = {l["product_id"][0]: l for l in found_lines}
         RecomendationLine = self.env["sale.order.recommendation.line"]
+        wiz_line_ids = self.env["sale.order.recommendation.line"]
         existing_product_ids = []
         # Add products from sale order lines
         for line in self.order_id.order_line:
             found_line = found_dict[line.product_id.id]
-            new_line = RecomendationLine.new({
+            new_line = RecomendationLine.create({
```
Comment (yajo) [inline sale_order_product_recommendation/wizards/sale_order_recommendation.py]: In general terms, creating records in an onchange is not a very good idea... This is a transient model, so records get removed every now and then, so maybe we can make an exception?

:thinking: Have you tried converting each `.new()` to `(0, 0, {...})`? I think that behind the scenes both get casted to the same operation, but @rco-odoo told me recently that this wasn't 100% the case. At least it's worth trying?

File: sale_order_product_recommendation/wizards/sale_order_recommendation.py [hunk 2]
```diff
@@ -80,42 +80,37 @@ def _generate_recommendations(self):
...
             if i == self.line_amount:
                 break
-
-    @api.model
-    def create(self, vals):
-        if 'line_ids' in vals:
-            vals['line_ids'] = [
-                line for line in vals['line_ids'] if line[2]["is_modified"]]
-        return super(SaleOrderRecommendation, self).create(vals)
+        self.line_ids = wiz_line_ids
```
Comment (yajo) [inline sale_order_product_recommendation/wizards/sale_order_recommendation.py]: Or is it possible that all those `+=` mess things up? Have you tried just storing all new records in a separate variable and assigning it just once at the end like here?

BTW you can remove line 56 with this approach.

File: sale_order_product_recommendation/wizards/sale_order_recommendation.py [hunk 3]
```diff
@@ -182,7 +178,6 @@ class SaleOrderRecommendationLine(models.TransientModel):
         "sale.order.recommendation",
         "Wizard",
         ondelete="cascade",
-        required=True,
```
Comment (yajo) [inline sale_order_product_recommendation/wizards/sale_order_recommendation.py]: This should be required...

File: sale_order_product_recommendation/wizards/sale_order_recommendation.py (hunk 1 above)
Comment (pedrobaeza) [reply sale_order_product_recommendation/wizards/sale_order_recommendation.py]: Of course, you shouldn't `create` on an onchange! @chienandalu this is not correct, so you have to find another way to solve the problem.

**Comment by pedrobaeza** [comment]
@Yajo, this is the correct solution

@chienandalu please review it.

**Review by chienandalu — COMMENTED** [review]
:+1: `force_save` was the key

**Review by rafaelbn — APPROVED** [review]

**Review by yajo — APPROVED** [review]

**Pull request merged**